vignettes/vignette_best_practices.Rmd

```diff
@@ -0,0 +1,198 @@
+---
+title: "Best practices when writing vignette in admiral"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Best practices when writing vignette in admiral}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup0, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This guidance ensures consistency across the vignettes in the admiral package in terms of content, structure and code execution. 
+As a general rule, the vignette workflow defined in [r-pkgs.org](https://r-pkgs.org/vignettes.html) is followed.
+
+
+# Metadata
+
+Each vignette in admiral should start with the following metadata.
+
+```
+---
+title: "<Title>"
+output:  rmarkdown::html_vignette:
+vignette: >
+  %\VignetteIndexEntry{<Title>}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+```
+
+The `<Title>` of the vignette should be meaningful. 
+
+# Markdown
+
+## Default Options
+
+    `r ''````{r setup, include=FALSE}
+    knitr::opts_chunk$set(
+      collapse = TRUE,
+      comment = "#>"
+      )
+    ```
+
+
+## Format Sections
+
+### Table of Contents
+
+Headings must be title case and start from Heading 1:
+
+```
+# Heading 1
+This is the description of my first section.
+
+## Heading 2.1
+This is the description of my first sub-section within my first section.
+
+## Heading 2.2
+This is the description of my second sub-section within my first section.
+
+```
+
+The first section gives a brief introduction of the vignette. The last sub-section 
+of the introduction should describe the packages required to run the admiral functions. The admiral package should be described first.
+
+```
+# Introduction
+
+This is the introduction of my mignette.
+
+## Required Packages
+
+    `r ''````{r, warning=FALSE, message=FALSE}
+    library(admiral)
+    # <all packages used in the vignette>
+    ```
+```
+
+The `warning = FALSE` and `message = FALSE` options are there to prevent the usual messages:
+
+<span style="color: red;font-family: textmate; font-size = 12pt;" >Attaching package: 'xxxx'</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">The following objects are masked from 'package:yyyyy':</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">  fun1, fun2</span>
+
+### Conventions
+
+#### General Conventions
+
++ Any variable name, dataset name, function, argument name must be quoted with backticks: e.g. '`'ADSL'`', '`'dm'`', '`'ADT'`'
```

Comment:
Perhaps add a bullet regarding casing.  I think we have used uppercase for ADaMs but I now some people would treat `ADSL` as lowercase.

Reply:
I have added "Variables name is expected to be uppercase while datasets name is expected to be lowercase." since this what seems to have been used in our vignettes. Is that OK?